server/routes/signup/signupRoutes.ts

```diff
@@ -1,16 +1,122 @@
-import { Router } from "express";
+import { Router, Request, Response } from "express";
+import { validationResult } from "express-validator";
+
 import Handler from "./signupHandler";
 
 const signupRouter = Router();
 
 const handler = new Handler();
 
-signupRouter.post("/parents", handler.signupValidation, handler.signupParent);
+signupRouter.post(
+  "/parents",
+  handler.parentsFieldValidation(),
+  async (req: Request, res: Response) => {
+    const fieldValidationResult = validationResult(req);
+    if (!fieldValidationResult.isEmpty()) {
+      console.log(fieldValidationResult.array()[0].msg);
+      return res
+        .status(400)
+        .json({ message: fieldValidationResult.array()[0].msg });
+    }
+
+    const {
+      name,
+      email,
+      password,
+      city,
+      street,
+      phoneNumber,
+      gender,
+      minKidAge,
+      maxKidAge,
+      numOfKids,
+      comments,
+    } = req.body;
+
+    try {
+      const validation = await handler.parentSignupValidation(email, password);
+
+      if (!validation.isValid) {
+        return res.status(400).send({ error: validation.message });
+      }
+
+      await handler.signupParent({
+        name,
+        email,
+        password,
+        city,
+        street,
+        phoneNumber,
+        gender,
+        minKidAge,
+        maxKidAge,
+        numOfKids,
+        comments,
+      });
+
+      return res.status(200).json({ message: "Signed up successfully" }).end();
+    } catch (error) {
+      const e = error as Error;
+      console.log(`Error message: ${req.body}: ${e.message}\n${e.stack}`);
+      return res.status(500).send();
+    }
+  }
+);
 
 signupRouter.post(
   "/babysitters",
-  handler.signupValidation,
-  handler.signupBabysitter
+  handler.babysitterFieldValidation(),
+  async (req: Request, res: Response) => {
+    const fieldValidationResult = validationResult(req);
+    if (!fieldValidationResult.isEmpty()) {
+      return res
+        .status(400)
+        .json({ message: fieldValidationResult.array()[0].msg });
+    }
+
+    const {
+      name,
+      email,
+      password,
+      city,
+      street,
+      experience,
+      age,
+      phoneNumber,
+      gender,
+      comments,
+    } = req.body;
+
+    try {
+      const validation = await handler.babysitterSignupValidation(
+        email,
+        password
+      );
+
+      if (!validation.isValid) {
+        return res.status(400).send({ error: validation.message });
+      }
+
+      await handler.signupBabysitter({
+        name,
+        email,
+        password,
+        city,
+        street,
+        experience,
+        age,
+        phoneNumber,
+        gender,
+        comments,
+      });
+
+      return res.status(200).json({ message: "Signed up successfully" }).end();
```

Comment:
you can also not pass anything in the json and if no error was returned then is obvious that it past

Reply:
don't you want a message to indicate the status in addition to the code?

Reply:
it's ok you can leave it, but just as FYI